hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -3311,6 +3311,25 @@ protected final void renderCommaSeparated(Iterable<? extends SqlAstNode> express
 		}
 	}
 
+	protected final void renderNonNullCommaSeparated(Iterable<? extends SqlAstNode> expressions) {
+		String separator = NO_SEPARATOR;
+		for ( SqlAstNode expression : expressions ) {
+			if( !isNullValue( expression ) ){
+				appendSql( separator );
+				expression.accept( this );
+				separator = COMA_SEPARATOR;
+			}
+		}
+	}
+
+	private boolean isNullValue(SqlAstNode expression) {
+		if(expression instanceof QueryLiteral){
+			QueryLiteral literal = (QueryLiteral) expression;
+			return literal.getLiteralValue().equals( "null" ) || literal.getLiteralValue().equals( "not null" );
```

Comment:
Wouldn't this also be triggered if I did a query like `from Entity a where a.name in ('null', 'not null')` which is completely unrelated? IMO we should introduce special `QueryLiteral` objects or a special expression class for which we can do an identity check here.